packages/yourloops/components/chat/chat-widget.tsx

```diff
@@ -0,0 +1,231 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+
+import Picker, { IEmojiData } from "emoji-picker-react";
+
+import { makeStyles, Theme } from "@material-ui/core/styles";
+import SendIcon from "@material-ui/icons/Send";
+import SentimentSatisfiedOutlinedIcon from "@material-ui/icons/SentimentSatisfiedOutlined";
+import EmailOutlinedIcon from "@material-ui/icons/EmailOutlined";
+import Card from "@material-ui/core/Card";
+import ChatMessage from "./chat-message";
+import { getChatMessages, sendChatMessage } from "../../lib/chat/api";
+import { useAuth } from "../../lib/auth";
+import { IMessage } from "../../models/chat";
+
+const chatWidgetStyles = makeStyles((theme: Theme) => {
+  return {
+    chatWidget: {
+      width: "300px",
+      height: "500px",
+    },
+    chatWidgetHeader: {
+      padding: theme.spacing(1),
+      display: "flex",
+      alignItems: "center",
+      background: "#035271",
+      color: "black",
+      width: "100%",
+      height: "40px",
+    },
+    chatWidgetHeaderText: {
+      padding: theme.spacing(1),
+    },
+    iconButton: {
+      "backgroundColor": "white",
+      "border": "none",
+      "& :hover": {
+        cursor: "pointer",
+      },
+    },
+    chatWidgetContent: {
+      background: "white",
+      width: "100%",
+      height: "420px",
+      overflow: "auto",
+    },
+    chatWidgetFooter: {
+      display: "flex",
+      alignItems: "center",
+      borderTop: "1px solid #E9E9E9",
+      background: "white",
+      borderRadius: "0px 0px 12px 12px",
+      width: "100%",
+      height: "40px",
+      paddingTop: theme.spacing(1),
+      paddingBottom: theme.spacing(1),
+    },
+    chatWidgetInput: {
+      "fontFamily": "Roboto",
+      "fontStyle": "normal",
+      "marginLeft": theme.spacing(1),
+      "display": "block",
+      "width": "90%",
+      "maxWidth": "90%",
+      "overflow": "hidden",
+      "resize": "none",
+      "height": "24px",
+      "maxHeight": "60px",
+      "lineHeight": "20px",
+      "border": "1px solid white",
+      "&:focus" : {
+        outlineColor: "#F6F6F6",
+      },
+    },
+    emojiPicker: {
+      display: "none",
+    },
+    icon: {
+      margin: theme.spacing(1),
+    },
+  };
+}, { name: "ylp-chat-widget" });
+
+export interface Message {text: string, isMine: boolean}
+export interface ChatWidgetProps {
+  patientId: string;
+  teamId: string;
+  userId: string
+  role: string
+}
+
+function ChatWidget(props: ChatWidgetProps): JSX.Element {
```

Comment:
I think that it should be possible to write some unit tests for this component :slightly_smiling_face:

Reply:
done